biodatasets/nagel/nagel.py

```diff
@@ -0,0 +1,312 @@
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+OTHER DEALINGS IN THE SOFTWARE.
+"""
+
+_URLS = {
+    _DATASETNAME: "https://sourceforge.net/projects/bionlp-corpora/files/ProteinResidue/NagelCorpus.tar.gz/download",
+}
+
+_SUPPORTED_TASKS = [Tasks.NAMED_ENTITY_RECOGNITION]  # , Tasks.RELATION_EXTRACTION]
+
+_SOURCE_VERSION = "1.0.0"
+_BIGBIO_VERSION = "1.0.0"
+
+
+class NagelDataset(datasets.GeneratorBasedBuilder):
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = [
+        BigBioConfig(
+            name="nagel_source",
+            version=SOURCE_VERSION,
+            description="nagel source schema",
+            schema="source",
+            subset_id="nagel",
+        ),
+        BigBioConfig(
+            name="nagel_bigbio_kb",
+            version=BIGBIO_VERSION,
+            description="Nagel BigBio schema",
+            schema="bigbio_kb",
+            subset_id="nagel",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "nagel_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "document_id": datasets.Value("string"),
+                    "text": datasets.Value("string"),
+                    "xml_annotated_text": datasets.Value("string"),
+                    "entities": [
+                        {
+                            "offsets": [datasets.Value("int64")],
+                            "type": datasets.Value("string"),
+                            "AminoAcid_WildType_3_Letter_Abbrev": datasets.Value(
+                                "string"
+                            ),
+                            "Residue_Position": datasets.Value("string"),
+                            "AminoAcid_MutatedType_3_Letter_Abbrev": datasets.Value(
+                                "string"
+                            ),
+                            "Residue_mention_in_original_text": datasets.Value(
+                                "string"
+                            ),
+                        }
+                    ],
+                }
+            )
+
+        elif self.config.schema == "bigbio_kb":
+            features = schemas.kb_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+
+        urls = _URLS[_DATASETNAME]
```

Comment:
style related, could you replace `urls` to singular as it looks like this dataset uses only one URL?

```suggestion
        url = _URLS[_DATASETNAME]
```

Reply:
fixed + with other references to url